lib/texture2.c

```diff
@@ -2556,25 +2572,27 @@ ktxTexture2_inflateZstdInt(ktxTexture2* This, ktx_uint8_t* pDeflatedData,
         levelOffset += paddedLevelByteLength;
         inflatedDataCapacity -= paddedLevelByteLength;
     }
-    ZSTD_freeDCtx(dctx);
 
     // Now modify the texture.
 
     This->dataSize = inflatedByteLength;
     This->supercompressionScheme = KTX_SS_NONE;
     memcpy(cindex, nindex, levelIndexByteLength); // Update level index
-    free(nindex);
     This->_private->_requiredLevelAlignment = uncompressedLevelAlignment;
     // Set bytesPlane as we're now sized.
     uint32_t* bdb = This->pDfd + 1;
     // blockSizeInBits was set to the inflated size on file load.
     bdb[KHR_DF_WORD_BYTESPLANE0] = prtctd->_formatSize.blockSizeInBits / 8;
 
-    return KTX_SUCCESS;
-
 cleanup:
-    ZSTD_freeDCtx(dctx);
-    free(nindex);
+    if(dctx) {
+        ZSTD_freeDCtx(dctx);
+        dctx = NULL;
+    }
+    if(nindex) {
```

Comment:
Another tool (can't remember which) told us it is unnecessary to check for null pointers before calling `free`. So the `if(nidex)` is not needed. Please remove it.

I think the same is true of `ZSTD_freeDCtx`.